src/libslic3r/ExtrusionEntity.hpp

```diff
@@ -95,12 +95,18 @@ enum ExtrusionLoopRole : uint16_t {
 
 inline bool is_perimeter(ExtrusionRole role)
 {
-    return role == erPerimeter
+  return role == erPerimeter
         || role == erExternalPerimeter
         || role == erThinWall
         || role == erOverhangPerimeter;
 }
 
+inline bool is_external_perimeter(ExtrusionRole role)
+{
+    return role == erExternalPerimeter
+        || role == erThinWall;
+}
+
```

Comment:
same as for is_top_infill, don't add things you don't use.

Reply:
This was WIP. Will be removed